### PR TITLE
Add new snippet for removing a contact with the given email address

### PR DESCRIPTION
**Pull request by ko1in1u**
This change is [Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/204)

### CODE REVIEW

**Comment by ko1in1u** [comment]
API now supports deleting contacts under specific accounts

**Review by xpconanfan — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/204)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @ko1in1u)

___
*[`src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java` line 41 at r1](https://reviewable.io/reviews/google/mobly-bundled-snippets/204#-O4gumpcEzO_o153pkXm:-O4gumpcEzO_o153pkXn:boyv7ao) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/242734fb2498513ebd3f92b1ebade285818a702b/src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java#L41)):*
> ```Java
> 
>   @Rpc(description =
>       "Add a contact with the given email address. A Google account need to be specified, then the"
> ```

Do we mean "Add a contact to a Google account on the device"?

e.g. what will happen if an account that doesn't exist on the device is specified?
___
*[`src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java` line 67 at r1](https://reviewable.io/reviews/google/mobly-bundled-snippets/204#-O4gv1Q66h6SHpk-1yMJ:-O4gv1Q66h6SHpk-1yMK:buqrmcs) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/242734fb2498513ebd3f92b1ebade285818a702b/src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java#L67)):*
> ```Java
>   }
> 
>   @Rpc(description = "Remove a contact with the given email address.")
> ```

Do we mean "Remove a contact from a Google account on the device"?

**Comment by ko1in1u** [comment]
> > _[Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/204)_ status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @ko1in1u)
> 
> _[`src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java` line 41 at r1](https://reviewable.io/reviews/google/mobly-bundled-snippets/204#-O4gumpcEzO_o153pkXm:-O4gumpcEzO_o153pkXn:boyv7ao) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/242734fb2498513ebd3f92b1ebade285818a702b/src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java#L41)):_
> 
> > ```java
> >   @Rpc(description =
> >       "Add a contact with the given email address. A Google account need to be specified, then the"
> > ```
> 
> Do we mean "Add a contact to a Google account on the device"?
> 
> e.g. what will happen if an account that doesn't exist on the device is specified?
> 

The Android API won't throw an error if account not exist, it will save as a device-only contact directly. I can call AccountSnippet.listAccounts function if we plan to throw an error for this case.

**Review by xpconanfan — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/204)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @ko1in1u)

___
*[`src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java` line 41 at r2](https://reviewable.io/reviews/google/mobly-bundled-snippets/204#-O4qw9Wj7BVuTGbtXQyS:-O4qw9Wj7BVuTGbtXQyT:b-uqipg9) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/8ae5228a59d1c3bde3ba0ee3cb2c8047d8c9abcc/src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java#L41)):*
> ```Java
> 
>   @Rpc(description = "Add a contact to a Google account on the device.")
>   public void addGoogleContact(String contactEmailAddress, String accountEmailAddress)
> ```

contactAddToGoogleAccountByEmail
contactRemoveFromGoogleAccountByEmail

**Review by xpconanfan — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/204)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @ko1in1u)

___
*[`src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java` line 41 at r1](https://reviewable.io/reviews/google/mobly-bundled-snippets/204#-O4gumpcEzO_o153pkXm:-O4qwV47A2QJcgHAvw3s:b-mvg3k6) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/242734fb2498513ebd3f92b1ebade285818a702b/src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java#L41)):*
<details><summary><i>Previously, ko1in1u (Kolin Lu) wrote…</i></summary><blockquote>

The Android API won't throw an error if account not exist, it will save as a device-only contact directly. I can call AccountSnippet.listAccounts function if we plan to throw an error for this case.
</blockquote></details>

sg, let's add a check for that

**Comment by ko1in1u** [comment]
> > _[Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/204)_ status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @ko1in1u)
> 
> _[`src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java` line 41 at r1](https://reviewable.io/reviews/google/mobly-bundled-snippets/204#-O4gumpcEzO_o153pkXm:-O4qwV47A2QJcgHAvw3s:b-mvg3k6) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/242734fb2498513ebd3f92b1ebade285818a702b/src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java#L41)):_
> 
> Previously, ko1in1u (Kolin Lu) wrote…
> sg, let's add a check for that

Done

File: src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java [hunk 1]
```diff
@@ -73,6 +98,39 @@ public void syncGoogleContacts(String accountEmailAddress) {
...
+      if (cursor != null && cursor.moveToFirst()) {
+        return cursor.getLong(
+            cursor.getColumnIndex(ContactsContract.CommonDataKinds.Email.CONTACT_ID));
+      }
+      throw new OperationApplicationException(
+          "The contact " + emailAddress + " doesn't appear to be saved on " + accountEmailAddress);
+    }
+  }
+
+  private void isAccountExist(String emailAddress) throws ContactSnippetException {
```
Comment (xpconanfan) [inline src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java:123]: assertAccountExists

Comment (ko1in1u) [reply src/main/java/com/google/android/mobly/snippet/bundled/ContactSnippet.java]: Done.

**Review by xpconanfan — APPROVED** [review]

**Pull request merged**